central/notifiers/jira/jira.go

```diff
@@ -277,7 +388,56 @@ func newJira(notifier *storage.Notifier, metadataGetter notifiers.MetadataGetter
 		return nil, err
 	}
 
-	decCreds := notifier.GetJira().GetPassword()
+	client, err := createClient(notifier, cryptoCodec, cryptoKey)
+
+	if err != nil {
+		return nil, err
+	}
+
+	canCreateIssues, err := canCreateIssuesInProject(client, notifier.GetLabelDefault())
+
+	if err != nil {
+		return nil, err
+	}
+
+	if !canCreateIssues {
+		return nil, fmt.Errorf("Cannot create issues in project %s", notifier.GetLabelDefault())
+	}
+
+	var priorityMapping map[storage.Severity]string
+	if !conf.DisablePriority {
+		priorityMapping, err = configurePriority(client, conf, notifier.GetLabelDefault())
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// marshal unknowns
+	var unknownMap map[string]interface{}
```

Comment:
How does this work with the below? Do the unmarshal allocate the map? I would have thought we needed to unknownMap := make(map[string]interface{})

Reply:
apparently not. I just tested the code, and it works as intended.  Here's a Go playground example that shows it working: https://go.dev/play/p/GU3-NB6WmZi.  I admit though that I'm not sure exactly how it works.

Reply:
Interesting!

Reply:
Did a bit of digging, and it looks like the JSON decoder will allocate an empty map if the pointer is to a nil map: https://github.com/golang/go/blob/go1.21.5/src/encoding/json/decode.go#L72